kamon-core/src/main/scala/kamon/trace/MetricsOnlyContext.scala

```diff
@@ -31,6 +31,7 @@ private[kamon] class MetricsOnlyContext(traceName: String, val token: String, iz
 
   @volatile private var _name = traceName
   @volatile private var _isOpen = izOpen
+  @volatile private var _isFinishedWithError = false
```

Comment:
don't add this variable, this extra bit of state is not necessary since we could just add a parameter to finish that says "this finished as error" and have the previous parameter-less version of finish() to call the new assuming the trace was successful.